### PR TITLE
Add Golang translator

### PR DESCRIPTION
**Pull request by simonz-bq**
*Issue #, if available:*

https://bitquill.atlassian.net/browse/AN-972

*Description of changes:*

Added GoTranslator for Java. Contains some TODOs for features not yet implemented in Gremlin-Go.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/process/traversal/translator/GolangTranslator.java
Comment (lyndonbauto) [inline gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/process/traversal/translator/GolangTranslator.java:54]:
```suggestion
 * Props to the author below.
 * @author Simon Zhao (simonz@bitquilltech.com)
```

File: gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/process/traversal/translator/GolangTranslator.java
Comment (lyndonbauto) [inline gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/process/traversal/translator/GolangTranslator.java:153]: Does the framework surrounding this accept NotImplementedException and skip it, or is this going to cause it to crash (or at least exit) at runtime?

File: gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/process/traversal/translator/GolangTranslator.java
Comment (lyndonbauto) [inline gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/process/traversal/translator/GolangTranslator.java]: Sort of annoying, but can you make a separate ticket to address this? Rithin's Set implementation is already in review and won't have a chance to update this before merge.

File: gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/process/traversal/translator/GolangTranslator.java
Comment (lyndonbauto) [inline gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/process/traversal/translator/GolangTranslator.java]:
```suggestion
}

```

File: gremlin-core/src/test/java/org/apache/tinkerpop/gremlin/process/traversal/translator/GolangTranslatorTest.java
Comment (lyndonbauto) [inline gremlin-core/src/test/java/org/apache/tinkerpop/gremlin/process/traversal/translator/GolangTranslatorTest.java]: Have you run these / do they pass?

File: gremlin-core/src/test/java/org/apache/tinkerpop/gremlin/process/traversal/translator/GolangTranslatorTest.java
Comment (lyndonbauto) [inline gremlin-core/src/test/java/org/apache/tinkerpop/gremlin/process/traversal/translator/GolangTranslatorTest.java]:
```suggestion
    public void shouldTranslateGoNamedSteps() {
```

File: gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/process/traversal/translator/GolangTranslator.java
Comment (simonz-bq) [reply gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/process/traversal/translator/GolangTranslator.java:153]: NotImplementedException is a RuntimeException so it will exit.

File: gremlin-core/src/test/java/org/apache/tinkerpop/gremlin/process/traversal/translator/GolangTranslatorTest.java
Comment (L0Lmaker) [inline gremlin-core/src/test/java/org/apache/tinkerpop/gremlin/process/traversal/translator/GolangTranslatorTest.java:118]: Whats the coverage of these tests?

File: gremlin-core/src/test/java/org/apache/tinkerpop/gremlin/process/traversal/translator/GolangTranslatorTest.java
Comment (simonz-bq) [reply gremlin-core/src/test/java/org/apache/tinkerpop/gremlin/process/traversal/translator/GolangTranslatorTest.java]: Yes, the tests pass except for the disabled ones.

File: gremlin-core/src/test/java/org/apache/tinkerpop/gremlin/process/traversal/translator/GolangTranslatorTest.java
Comment (simonz-bq) [reply gremlin-core/src/test/java/org/apache/tinkerpop/gremlin/process/traversal/translator/GolangTranslatorTest.java:118]: Whatever the coverage of the Python tests are.

**Review by lyndonbauto — APPROVED** [review]
LGTM, thanks @simonz-bq. Should probably wait for @xiazcy to review before merging since the thing she is working on is related and she may have useful input.

**Pull request merged**
